src/CubedSpheres/cubed_sphere_halo_filling.jl

```diff
@@ -0,0 +1,224 @@
+    cubed_sphere_halo(u, u_loc, 2, :south) .= - reverse(permutedims(cubed_sphere_boundary(v, v_loc, 6, :east),  (2, 1, 3)), dims=1)
+    cubed_sphere_halo(u, u_loc, 2, :north) .=                       cubed_sphere_boundary(u, u_loc, 3, :south)
+
+    # Face 2 v-velocity
+    cubed_sphere_halo(v, v_loc, 2, :west)  .=                       cubed_sphere_boundary(v, v_loc, 1, :east)
+    cubed_sphere_halo(v, v_loc, 2, :east)  .= - reverse(permutedims(cubed_sphere_boundary(u, u_loc, 4, :south), (2, 1, 3)), dims=2)
+    cubed_sphere_halo(v, v_loc, 2, :south) .= + reverse(permutedims(cubed_sphere_boundary(u, u_loc, 6, :east),  (2, 1, 3)), dims=1)
+    cubed_sphere_halo(v, v_loc, 2, :north) .=                       cubed_sphere_boundary(v, v_loc, 3, :south)
+
+    # Face 3 u-velocity
+    cubed_sphere_halo(u, u_loc, 3, :west)  .= + reverse(permutedims(cubed_sphere_boundary(v, v_loc, 1, :north), (2, 1, 3)), dims=2)
+    cubed_sphere_halo(u, u_loc, 3, :east)  .=                       cubed_sphere_boundary(u, u_loc, 4, :west)
+    cubed_sphere_halo(u, u_loc, 3, :south) .=                       cubed_sphere_boundary(u, u_loc, 2, :north)
+    cubed_sphere_halo(u, u_loc, 3, :north) .= - reverse(permutedims(cubed_sphere_boundary(v, v_loc, 5, :west),  (2, 1, 3)), dims=1)
+
+    # Face 3 v-velocity
+    cubed_sphere_halo(v, v_loc, 3, :west)  .= - reverse(permutedims(cubed_sphere_boundary(u, u_loc, 1, :north), (2, 1, 3)), dims=2)
+    cubed_sphere_halo(v, v_loc, 3, :east)  .=                       cubed_sphere_boundary(v, v_loc, 4, :west)
+    cubed_sphere_halo(v, v_loc, 3, :south) .=                       cubed_sphere_boundary(v, v_loc, 2, :north)
+    cubed_sphere_halo(v, v_loc, 3, :north) .= + reverse(permutedims(cubed_sphere_boundary(u, u_loc, 5, :west),  (2, 1, 3)), dims=1)
+
+    # Face 4 u-velocity
+    cubed_sphere_halo(u, u_loc, 4, :west)  .=                       cubed_sphere_boundary(u, u_loc, 3, :east)
+    cubed_sphere_halo(u, u_loc, 4, :east)  .= + reverse(permutedims(cubed_sphere_boundary(v, v_loc, 6, :south), (2, 1, 3)), dims=2)
+    cubed_sphere_halo(u, u_loc, 4, :south) .= - reverse(permutedims(cubed_sphere_boundary(v, v_loc, 2, :east),  (2, 1, 3)), dims=1)
+    cubed_sphere_halo(u, u_loc, 4, :north) .=                       cubed_sphere_boundary(u, u_loc, 5, :south)
+
+    # Face 4 v-velocity
+    cubed_sphere_halo(v, v_loc, 4, :west)  .=                       cubed_sphere_boundary(v, v_loc, 3, :east)
+    cubed_sphere_halo(v, v_loc, 4, :east)  .= - reverse(permutedims(cubed_sphere_boundary(u, u_loc, 6, :south), (2, 1, 3)), dims=2)
+    cubed_sphere_halo(v, v_loc, 4, :south) .= + reverse(permutedims(cubed_sphere_boundary(u, u_loc, 2, :east),  (2, 1, 3)), dims=1)
+    cubed_sphere_halo(v, v_loc, 4, :north) .=                       cubed_sphere_boundary(v, v_loc, 5, :south)
+
+    # Face 5 u-velocity
+    cubed_sphere_halo(u, u_loc, 5, :west)  .= + reverse(permutedims(cubed_sphere_boundary(v, v_loc, 3, :north), (2, 1, 3)), dims=2)
```

Comment:
@ali-ramadhan I am not sure this is correct?  Should u on face 5 have opposite sign to v on face 1?

Reply:
Yes it should, although the line you highlighted is looking at face 5 west and face 3 north exchange?

Reply:
Ooops!